adapters/zmaticoo/zmaticoo.go

```diff
@@ -0,0 +1,157 @@
+package zmaticoo
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"net/http"
+	"net/url"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the zmaticoo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	endpointURL, err := url.Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("invalid endpoint: %v", err)
+	}
+	bidder := &adapter{
+		endpoint: endpointURL.String(),
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+	adapterRequest, errs := a.makeRequest(request)
+	if errs == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	}
+	return adapterRequests, errs
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+	zmaticooExt, errs := getZmaticooExt(request)
+	if zmaticooExt == nil {
+		return nil, errs
+	}
```

Comment:
This code path is not covered. Requesting to add json test

Reply:
ok,i have changed this code